lib/studies/reply/adaptive_nav.dart

```diff
@@ -909,63 +916,99 @@ class _ReplyLogo extends StatelessWidget {
+      },
+      openColor: theme.cardColor,
+      closedShape: isDesktop & widget.extended
+          ? const StadiumBorder()
+          : const CircleBorder(),
+      closedColor: theme.colorScheme.secondary,
+      closedBuilder: (context, openContainer) {
+        if (isDesktop) {
+          return AnimatedSize(
+            vsync: this,
+            curve: Curves.easeInOut,
+            duration: kThemeAnimationDuration,
+            child: FloatingActionButton.extended(
+              heroTag: 'Rail FAB',
+              tooltip: widget.onMailView ? 'Reply' : 'Compose',
+              isExtended: widget.extended,
+              onPressed: openContainer,
+              label: Row(
+                children: [
+                  _FabSwitcher(
+                    onMailView: widget.onMailView,
+                  ),
+                  SizedBox(width: widget.extended ? 16 : 0),
+                  if (widget.extended)
+                    Text(
+                      widget.onMailView ? 'REPLY' : 'COMPOSE',
+                      style: Theme.of(context)
+                          .textTheme
+                          .headline5
+                          .copyWith(fontSize: 16, color: ReplyColors.black900),
+                    ),
+                ],
               ),
-          ],
-        ),
-      );
-    } else {
-      return FloatingActionButton(
-        heroTag: 'Bottom App Bar FAB',
-        tooltip: onMailView ? 'Reply' : 'Compose',
-        child: AnimatedSwitcher(
-          duration: const Duration(milliseconds: 350),
-          transitionBuilder: (child, animation) => ScaleTransition(
-            child: child,
-            scale: animation,
-          ),
-          child: onMailView
-              ? Icon(
-                  Icons.create,
-                  key: UniqueKey(),
-                )
-              : Icon(
-                  Icons.reply_all,
-                  key: UniqueKey(),
-                ),
-        ),
-        onPressed: () {
-          // TODO: Implement onPressed for Bottom App Bar FAB
-        },
-      );
-    }
+            ),
+          );
+        } else {
+          return FloatingActionButton(
+            heroTag: 'Bottom App Bar FAB',
+            tooltip: widget.onMailView ? 'Reply' : 'Compose',
+            child: _FabSwitcher(
+              onMailView: widget.onMailView,
+            ),
+            onPressed: openContainer,
+          );
+        }
+      },
+    );
+  }
+}
+
+class _FabSwitcher extends StatelessWidget {
+  const _FabSwitcher({@required this.onMailView}) : assert(onMailView != null);
+
+  final bool onMailView;
+
+  @override
+  Widget build(BuildContext context) {
+    return AnimatedSwitcher(
+      duration: const Duration(milliseconds: 350),
+      transitionBuilder: (child, animation) => ScaleTransition(
+        child: child,
+        scale: animation,
+      ),
+      child: onMailView
+          ? Icon(
+              Icons.reply_all,
+              key: UniqueKey(),
```

Comment:
nit: unsure if `key` is necessary, also can fit each `Icon` on 1 line

Reply:
`key` is necessary for the framework to differentiate between the two `Icon's`. If not then the framework won't animate between them when they switch, instead it will use the same  `Icon` instance but switch the parameters.